hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/CopyCommitter.java

```diff
@@ -152,9 +152,18 @@ public void abortJob(JobContext jobContext,
   }
 
   private void cleanupTempFiles(JobContext context) {
-    try {
-      Configuration conf = context.getConfiguration();
+    Configuration conf = context.getConfiguration();
+
+    final boolean directWrite = conf.getBoolean(
+        DistCpOptionSwitch.DIRECT_WRITE.getConfigLabel(), false);
+    final boolean append = conf.getBoolean(
+        DistCpOptionSwitch.APPEND.getConfigLabel(), false);
+    final boolean useTempTarget = !append && !directWrite;
+    if (!useTempTarget) {
```

Comment:
Reviewing this, I think the cleanup should only be skipped on direct *not* append.

why so, append will choose between overwrite and append depending on the state of the destination file -a decision made case by case. when doing distcp with -append, new files or those whose dest checksum doesn't match that of the source for the same #of bytes will use a temp file unless -direct is set.

Reply:
Isn't that same check as here in RetriableFileCopyCommand
```
    final boolean toAppend = action == FileAction.APPEND;
    final boolean useTempTarget = !toAppend && !directWrite;
    Path targetPath = useTempTarget ? getTempFile(target, context) : target;
```
https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java#L135-L137

Reply:
it's doing it file by file: some files will be appended to, but new ones, ones with checksum mismatch are overwritten by first writing to temp, then rename, as usual

any -append job which added files can create temp paths

Reply:
Thanx for the explanation. Will give some time to the original author, In case no one volunteers for few days, I will fix this.